File: 14th-April-Remove_the_balls/c++/solution.cpp
/*
  This code is written by Bhaskar
*/

class Solution {
  public:
    int finLength(int N, vector<int> color, vector<int> radius) {
        vector<int> col;
        vector<int> rad;

        col.push_back(color[0]);
        rad.push_back(radius[0]);

        for (int i = 1; i < N; i++) {
            if (!col.empty() && !rad.empty() && (color[i] == col.back()) && (radius[i] == rad.back())) {
                col.pop_back();
                rad.pop_back();
            } else {
                col.push_back(color[i]);
                rad.push_back(radius[i]);
            }
        }

        return col.size();
    }
};

File: 14th-April-Remove_the_balls/java/solution.java
/*
  This code is written by Bhaskar
*/

class Solution {
    public static int finLength(int N, int[] color, int[] radius) {
        // code here
        ArrayList<Integer> col = new ArrayList<>();
        ArrayList<Integer> rad = new ArrayList<>();
        
        col.add(color[0]);
        rad.add(radius[0]);
        
        for(int i=1 ; i< N ; i++){
            if(col.size()>0 && rad.size()>0 && (color[i]==col.get(col.size()-1)) && (radius[i]==rad.get(rad.size()-1))){
                col.remove(col.size()-1);
                rad.remove(rad.size()-1);
            }else{
                col.add(color[i]);
                rad.add(radius[i]);
            }
        }
        
        return col.size();
    }
}
